pkg/operators/wasm/maps.go

```diff
@@ -213,3 +270,26 @@ func (i *wasmOperatorInstance) mapDelete(ctx context.Context, m wapi.Module, sta
 	}
 	stack[0] = 0
 }
+
+// mapClose close the map.
+// Params:
+// - stack[0]: Map handle
+// Return value:
+// - 0 on success, 1 on error
+func (i *wasmOperatorInstance) mapClose(ctx context.Context, m wapi.Module, stack []uint64) {
+	mapHandle := wapi.DecodeU32(stack[0])
+
+	ebpfMap, ok := getHandle[*ebpf.Map](i, mapHandle)
+	if !ok {
+		stack[0] = 1
+		return
+	}
+
+	err := ebpfMap.Close()
```

Comment:
* I understand closing a map created by `newMap`, but how is it safe to close a map from `getMap`? For example, what if the wasm module closes the map "events" or the map "sockets" from the socket enricher? It seems like it will close the map without checking if it can be used internally by IG.
* How do you prevent a wasm program from calling Lookup after calling Close? It seems IG will call the bpf syscall with fd -1. Shouldn't we detect that?

Reply:
> Why do you mean by prevent?

I mean that IG should detect when mapClose has been called on a handle. Then, further calls to mapClose would return an error without relying on the kernel returning EBADF.

> Otherwise, you can call the BPF syscall with fd -1 and you will get an error as reported by the kernel.

True, but this looks weird to me. It is exposing low-level details to the wasm module instead of wrapping it.